src/nvim/tui/tui.c

```diff
@@ -1658,6 +1659,37 @@ static void out(void *ctx, const char *str, size_t len)
   data->bufpos += len;
 }
 
+/// Called by unibi_format() for padding.
+static void pad(void *ctx, size_t delay, int scale, int force)
```

Comment:
// TODO: the delay impl is mostly copied from loop_poll_events(). Can we extract this pattern to a common function?